rust/extractor/src/main.rs

```diff
@@ -65,16 +68,20 @@ fn extract(
         )
     });
 }
-fn main() -> anyhow::Result<()> {
-    let cfg = config::Config::extract().context("failed to load configuration")?;
+
+fn run_extractor(mut cfg: config::Config) -> anyhow::Result<()> {
     stderrlog::new()
         .module(module_path!())
-        .verbosity(1 + cfg.verbose as usize)
+        .verbosity(cfg.verbose as usize)
         .init()?;
+    if cfg.qltest {
+        qltest::prepare(&mut cfg)?;
+    }
+    info!("configuration: {cfg:#?}\n");
```

Comment:
Don't print the `cfg` at the info level. It's OK at trace level or similar.  It can be massive and renders log output very hard to view for large repositories. If you suppress printing the `cfg.inputs` field then it should be fine though.

Reply:
ah, true, the inputs are too much, the rest is really useful though. Let me look if there's an easy way to suppress a field in the debug print